frontend/src/services/userService.ts

```diff
@@ -0,0 +1,31 @@
+import api from '../common/helpers/apiHelper';
+import { IUser } from 'common/models/user/IUser';
+import { IForgotPasswordInput } from 'common/models/auth/IForgotPasswordInput';
+
+export interface IEditStatusData {
+  [key: string]: string;
+}
+type ServerResponse = string;
+
+export async function editStatus(editStatusData: IEditStatusData) {
+  // const response = await api.put<ServerResponse>('/edit-status', { editStatusData });
+  const response = editStatusData.status; // mocked
+  return response;
+}
+
+export async function deleteUser() {
+  await api.delete('/api/users/');
+}
+
+export async function editUser(data: IUser) {
+  const response = await api.put<IUser>('/api/users/', { data });
```

Comment:
```suggestion
  const response = await api.put<IUser>('/api/users', { data });
```

you don't need slash in the end, remove it everywhere